Wikipedia/Code/ArticleViewController.swift

```diff
@@ -551,17 +562,48 @@ class ArticleViewController: ViewController, HintPresenting {
     // MARK: Refresh
     
     @objc public func refresh() {
+        state = .reloading
         if !shouldPerformWebRefreshAfterScrollViewDeceleration {
             updateRefreshOverlay(visible: true)
         }
         shouldPerformWebRefreshAfterScrollViewDeceleration = true
     }
+    
+    /// Preserves the current scroll position, loads the provided revisionID or waits for a change in etag on the mobile-html response, then refreshes the page and restores the prior scroll position
+    internal func waitForNewContentAndRefresh(_ revisionID: UInt64? = nil) {
+        showNavigationBar()
+        state = .reloading
+        saveArticleScrollPosition()
+        isRestoringState = true
+        setupForStateRestorationIfNecessary()
+        // If a revisionID was provided, just load that revision
+        if let revisionID = revisionID {
+            performWebViewRefresh(revisionID)
+            return
+        }
+        // If no revisionID was provided, wait for the ETag to change
+        guard let eTag = currentETag else {
+            performWebViewRefresh()
+            return
+        }
+        fetcher.waitForMobileHTMLChange(articleURL: articleURL, eTag: eTag, maxAttempts: 5) { (result) in
+            DispatchQueue.main.async {
+                switch result {
+                case .failure(let error):
+                    self.showError(error)
+                    fallthrough
+                default:
+                    self.performWebViewRefresh()
+                }
+            }
+        }
+    }
 
-    internal func performWebViewRefresh() {
+    internal func performWebViewRefresh(_ revisionID: UInt64? = nil) {
         #if WMF_LOCAL_PAGE_CONTENT_SERVICE // on local PCS builds, reload everything including JS and CSS
             webView.reloadFromOrigin()
         #else // on release builds, just reload the page with a different cache policy
-            loadPage(cachePolicy: .noPersistentCacheOnError)
+            loadPage(cachePolicy: .noPersistentCacheOnError, revisionID: revisionID)
```

Comment:
I notice that reloading this page with the revisionID does not update the underlying saved page content. If you change to airplane mode after a content update, clear out URLCache.shared in the app settings and return, you will see the old content. My guess is because the additional revision ID in the path generates a different database key, so it doesn't see the saved article as being the same article to update. I'm not sure that this is a big enough deal to fix at this point in the release cycle because I confirmed a simple pull to refresh or revisiting the page with a connection updates the content with the latest.

Reply:
I made an attempt at fixing this with 0ad47a7 but wasn't able to fully test it yet - we can roll it back if you feel it's too invasive and too late for this kind of change.

Reply:
@joewalsh yeah, I think I would rather push this particular tweak to 6.6.1 to give us more testing time on it. I can make a Phab task on it here in a bit.

Reply:
Here's the phab [task](https://phabricator.wikimedia.org/T252435) for this bug.

Reply:
Possibly related to this, the only remaining functional quirk I'm noticing consistently is this scenario: 

1. Make an edit on an article. 
2. After publishing, observe the change and force quit the app.
3. Relaunch the app - state restoration takes you back to the same article.
4. Observe that the change from step 2 isn't visible.
5. Tapping edit again shows the most "up to date" edit.

![edit_stale](https://user-images.githubusercontent.com/5652327/81877838-0ce27c00-953b-11ea-8bd9-d2ebbf663b7f.gif)

@tonisevener Do you want to address that with this PR or save it for [T252435](https://phabricator.wikimedia.org/T252435) with 6.6.1 as well?

Reply:
@staykids oo good catch - I'm pretty certain the same fix would fix both problems. I'll ask @jtminor on what he thinks.

Reply:
@staykids Per [https://phabricator.wikimedia.org/T252435#6137802](url), this can wait till 6.6.1.